servers/audio_server.cpp

```diff
@@ -199,8 +199,8 @@ int AudioDriverManager::get_driver_count() {
 
 void AudioDriverManager::initialize(int p_driver) {
 	GLOBAL_DEF_RST("audio/driver/enable_input", false);
-	GLOBAL_DEF_RST("audio/driver/mix_rate", DEFAULT_MIX_RATE);
-	GLOBAL_DEF_RST("audio/driver/mix_rate.web", 0); // Safer default output_latency for web (use browser default).
+	GLOBAL_DEF_RST(PropertyInfo(Variant::INT, "audio/driver/mix_rate", PROPERTY_HINT_RANGE, "0,192000,1,or_greater,suffix:Hz"), DEFAULT_MIX_RATE);
```

Comment:
Is zero safe here? We do division by it in some places

Reply:
I've searched sources for this variable access and found that mix_rate = 0 means "use default". This is used for "mix_rate.web" for example. I didn't found other access to this option, at least with full path "audio/driver/mix_rate"

Here it is in "audio_server.cpp":
```cpp
int AudioDriver::_get_configured_mix_rate() {
	StringName audio_driver_setting = "audio/driver/mix_rate";
	int mix_rate = GLOBAL_GET(audio_driver_setting);

	// In the case of invalid mix rate, let's default to a sensible value..
	if (mix_rate <= 0) {
		WARN_PRINT(vformat("Invalid mix rate of %d, consider reassigning setting \'%s\'. \nDefaulting mix rate to value %d.",
				mix_rate, audio_driver_setting, AudioDriverManager::DEFAULT_MIX_RATE));
		mix_rate = AudioDriverManager::DEFAULT_MIX_RATE;
	}

	return mix_rate;
}
```

Reply:
As I linked too 🙂, however it's treated as an error and will show a warning, so unsure how correct it is

Reply:
It seems you're right, this function was introduced as a bug fix for a crash when mix_rate is not valid: [commit](https://github.com/godotengine/godot/commit/5a08091168782a924cd3a23baf31c5cd4cd63906).

For web build it seems that it supposed to rely on zero default value (from [this](https://github.com/godotengine/godot/commit/d187bb4e11182ee3edbc0008dba5762148bc530d) commit) to fallback to browser defaults (in [library_godot_audio.js](https://github.com/godotengine/godot/blob/d443f127ef1db9fb2fe623ba1c17ded5402aff10/platform/web/js/libs/library_godot_audio.js#L41)), but if I understood correctly it still uses same `_get_configured_mix_rate()` method to obtain this value. If this true, then it should never get zero and also should always complain "Invalid mix rate" in console. But I never used Godot for web so can't check.

Reply:
The warning should be silenced on the Web platform, as a mix rate project setting of `0` is valid there.

Reply:
Done. For non-web limit now starts from 1.

Reply:
> Done. For non-web limit now starts from 1.

The lowest mix rate could probably be set to 11025 Hz, as I don't think lower mix rates will work correctly at all.

Reply:
> The lowest mix rate hint could probably be set to 11025 Hz, as I don't think lower mix rates will work correctly at all.

Well, talking about this, will non standard rates work at all? I just tried few experiments and it seems that rates below 11025 has no difference with 11025. I've tried 8000, 4000, 1000 and 1. May be it is my audio card specific, or just it always rounded to nearest 11024 * N and 8000 * N values (I used 8000 because Audacity lists it as an one of standard rates and also 48k, 96k are multiplies of it).

Reply:
> Well, talking about this, will non standard rates work at all?

It depends on the audio driver, but I'd assume rates lower than 11025 Hz won't work on all platforms Godot supports.